File: Java/program49.java
///////////////////////////////////////////////////////
//
//  File name :     program49.java
//  Descreption :   Separate Digit and Count Even Digits from Input Number
//  Author :        Vaibhav
//  Date :          20/05/2025
//
///////////////////////////////////////////////////////

import java.util.Scanner;

class CountEven
{
    int iValue;

    int Display()
    {   
        int iDigit = 0;
        int iCnt = 0;
        
        while(iValue != 0)
        {   
            iDigit = iValue % 10; 
            if((iDigit % 2) == 0)
            {
                iCnt++;
            }
            iValue = iValue / 10;
        }
        return iCnt;
    }
}   
class program49
{
    public static void main(String A[])
    {
        int iNo = 0;
        int iRet = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter Number : ");
        iNo = sobj.nextInt();

        System.out.println("-----------------------------------------------------------------------------------------------");
        System.out.println("Number Received is "+iNo);
        System.out.println("-----------------------------------------------------------------------------------------------");
        

        CountEven cobj = new CountEven();

        cobj.iValue = iNo;

        iRet = cobj.Display();

        System.out.println("Number of Even Digits in Number "+iNo+" is "+iRet);

    }
}

File: program49.c
///////////////////////////////////////////////////////
//
//  File name :     program49.c
//  Descreption :   Separate Digit and Count Even Digits from Number
//  Author :        Vaibhav
//  Date :          12/05/2025
//
///////////////////////////////////////////////////////

#include<stdio.h>

int CountEven(int iNo)
{
    int iCnt = 0;
    int iDigit = 0;

    while(iNo != 0)
    {   
        iDigit = iNo % 10;
        iNo = iNo / 10;
        if(iDigit % 2 == 0)
        {
            iCnt++;
        }
    }
    return iCnt;
}

int main()
{
    int iValue = 0;
    int iRet = 0;
    
    printf("Enter number: \n");
    scanf("%d",&iValue);

    printf("-----------------------------------------------------------------------------------------------\n");
    printf("Number Received is %d\n",iValue);
    printf("-----------------------------------------------------------------------------------------------\n");

    iRet = CountEven(iValue);

    printf("Number of Digits are : %d\n",iRet);
    
    return 0;
}